packages/transaction-controller/src/TransactionController.ts

```diff
@@ -387,6 +388,30 @@ export class TransactionController extends BaseController<
     });
   }
 
+  /**
+   * Creates approvals for all unapproved transactions persisted.
+   */
+  initApprovals() {
+    const {
+      networkId,
+      providerConfig: { chainId },
+    } = this.getNetworkState();
+    const unapprovedTxs = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transactionMatchesNetwork(transaction, chainId, networkId),
+    );
+
+    Object.values(unapprovedTxs).forEach((txMeta) => {
```

Comment:
Very minor, but is it cleaner to use a standard `for of` loop here to avoid the callbacks?